### ISSUE TITLE
Documentation builds are failing due to sphinx-gallery changes

### ISSUE THREAD

**Issue by QuLogic** [matplotlib/matplotlib#8520]
See, for example, both doc builds in [this cron build](https://travis-ci.org/matplotlib/matplotlib/builds/223982985). They are both failing due to this warning:
```
Warning, treated as error:
WARNING:
========
Sphinx-Gallery now requires you to set the configuration variable
'backreferences_dir' in your config to activate the
backreferences. That is mini galleries clustered by the functions used
in the example scripts. Have a look at it in sphinx-gallery
http://sphinx-gallery.readthedocs.io/en/stable/index.html#examples-using-numpy-linspace
If you don't care about this features set in your conf.py
'backreferences_dir': False
```

Ping @choldgraf.

**Comment by choldgraf** [comment]
yeah this is because SG just switched their version on pip. I'll add a quick PR patch.

**Comment by QuLogic** [comment]
Fixed by #8522.
